File: crop/src/main/java/com/googy/crop/CornerHandleHelper.java
package com.googy.crop;

import android.graphics.Rect;

class CornerHandleHelper extends HandleHelper {
    CornerHandleHelper(Edge horizontalEdge, Edge verticalEdge) {
        super(horizontalEdge, verticalEdge);
    }

    void updateCropWindow(float x, float y, float targetAspectRatio, Rect imageRect, float snapRadius) {
        EdgePair activeEdges = this.getActiveEdges(x, y, targetAspectRatio);
        Edge primaryEdge = activeEdges.primary;
        Edge secondaryEdge = activeEdges.secondary;
        primaryEdge.adjustCoordinate(x, y, imageRect, snapRadius, targetAspectRatio);
        secondaryEdge.adjustCoordinate(targetAspectRatio);
        if (secondaryEdge.isOutsideMargin(imageRect, snapRadius)) {
            secondaryEdge.snapToRect(imageRect);
            primaryEdge.adjustCoordinate(targetAspectRatio);
        }

    }
}

File: app/src/main/java/com/googy/videoeditor/ui/crop/cropview/window/handle/CornerHandleHelper.kt
package com.googy.videoeditor.ui.crop.cropview.window.handle

import android.graphics.Rect
import com.googy.videoeditor.ui.crop.cropview.window.edge.Edge

internal class CornerHandleHelper(horizontalEdge: Edge, verticalEdge: Edge) : HandleHelper(horizontalEdge, verticalEdge) {

    override fun updateCropWindow(x: Float, y: Float, targetAspectRatio: Float, imageRect: Rect, snapRadius: Float) {
        val (primaryEdge, secondaryEdge) = this.getActiveEdges(x, y, targetAspectRatio)
        primaryEdge?.adjustCoordinate(x, y, imageRect, snapRadius, targetAspectRatio)
        secondaryEdge?.adjustCoordinate(targetAspectRatio)
        if (secondaryEdge?.isOutsideMargin(imageRect, snapRadius) == true) {
            secondaryEdge.snapToRect(imageRect)
            primaryEdge?.adjustCoordinate(targetAspectRatio)
        }
    }
}